ocaml/utils/clflags.ml

```diff
@@ -376,23 +376,31 @@ let set_dumped_pass s enabled =
     dumped_passes_list := dumped_passes
   end
 
-type extension = Comprehensions
+module Extension = struct
+  type t = Comprehensions
 
-let extensions = ref ([] : extension list) (* -extensions *)
-let set_standard () = extensions := []
+  let all = [ Comprehensions ]
 
-let string_of_extension = function
-| Comprehensions -> "comprehensions"
+  let extensions = ref ([] : t list)   (* -extension *)
+  let equal Comprehensions Comprehensions = true
 
-let extension_of_string = function
-| "comprehensions" -> Comprehensions
-| extn ->  raise (Arg.Bad(Printf.sprintf "Extension %s is not known" extn))
+  let disable_all_extensions = ref false             (* -no-extensions *)
+  let disable_all () = disable_all_extensions := true
 
-let add_extension extn =
-  let extension = extension_of_string (String.lowercase_ascii extn) in
-  extensions := extension::!extensions
+  let to_string = function
+    | Comprehensions -> "comprehensions"
+
+  let of_string = function
+    | "comprehensions" -> Comprehensions
+    | extn ->  raise (Arg.Bad(Printf.sprintf "Extension %s is not known" extn))
 
-let is_extension_enabled ext = List.mem ext !extensions
+  let enable extn =
+    let t = of_string (String.lowercase_ascii extn) in
```

Comment:
Do we want `String.lowercase_ascii` here or in `of_string`?

Reply:
I put it here, to keep to_string/from_string simple and symmetric. Let's keep it as is until another use comes up, ok?